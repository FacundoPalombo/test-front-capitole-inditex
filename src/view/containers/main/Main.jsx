import React from 'react';
import { Link, Outlet } from 'react-router-dom';
import { BounceLoader } from 'react-spinners';

import styles from './styles.module.scss';
import { useIsFetching } from '@tanstack/react-query';

const Main = () => {
  const isLoading = useIsFetching() > 0;

  return (
    <main id="main" data-testid="main">
      <nav
        id="nav"
        className={styles.navigation}
        aria-label="Navigate back to list of channels"
      >
        <Link
          className={styles.navigation__linkBack}
          aria-label="Go back to home"
          tabIndex="1"
          to="/"
        >
          <h2 className={styles.navigation__title}>Podcaster</h2>
        </Link>
        <div className={styles.navigation__loader}>
          {isLoading && (
            <BounceLoader
              size={40}
              color="#3976a8"
              loading={true}
              speedMultiplier={1.25}
            />
          )}
        </div>
      </nav>
      <hr className={styles.navigation__separator} />
      <Outlet />
    </main>
  );
};

export default Main;
