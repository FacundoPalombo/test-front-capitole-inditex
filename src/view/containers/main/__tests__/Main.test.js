import React from 'react';
import Main from '../Main';
import renderWithRouter from '../../../../utils/tests/renderWithRouter';

jest.mock(
  'react-spinners',
  () =>
    ({ loading }) =>
      loading ? 'Spinner Component' : ''
);

describe('Main unit test', () => {
  afterAll(jest.clearAllMocks);
  it('should render ok with props', () => {
    const { asFragment } = renderWithRouter(<Main />, {
      route: '/podcast/1234',
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
