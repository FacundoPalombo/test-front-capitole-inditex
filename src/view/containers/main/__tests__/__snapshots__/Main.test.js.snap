// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main unit test should render ok with props, when is loading 1`] = `
<DocumentFragment>
  <main
    data-testid="main"
    id="main"
  >
    <nav
      aria-label="Navigate back to list of channels"
      class="navigation"
      id="nav"
    >
      <a
        href="/"
      >
        <h2
          class="navigation__title"
        >
          Podcaster
        </h2>
      </a>
      <div
        class="navigation__loader"
      />
    </nav>
    <hr
      class="navigation__separator"
    />
    <div>
      MockOutlet
    </div>
  </main>
</DocumentFragment>
`;

exports[`Main unit test should render ok with props, when loading is false 1`] = `
<DocumentFragment>
  <main
    data-testid="main"
    id="main"
  >
    <nav
      aria-label="Navigate back to list of channels"
      class="navigation"
      id="nav"
    >
      <a
        href="/"
      >
        <h2
          class="navigation__title"
        >
          Podcaster
        </h2>
      </a>
      <div
        class="navigation__loader"
      >
        <div>
          Loading...
        </div>
      </div>
    </nav>
    <hr
      class="navigation__separator"
    />
    <div>
      MockOutlet
    </div>
  </main>
</DocumentFragment>
`;
